postgraphile/website/postgraphile/usage-schema.md

```diff
@@ -212,20 +212,20 @@ automatically reflect the correct types thanks to graphql-codegen:
 
 ```ts
 import type { DocumentNode, ExecutionResult } from "postgraphile/graphql";
-import type { TypedDocumentNode } from "@graphql-typed-document-node/core";
+import type { TypedDocumentNode, VariablesOf, ResultOf } from "@graphql-typed-document-node/core";
 import { postgraphile } from "postgraphile";
 import { execute, hookArgs } from "postgraphile/grafast";
 import { validate } from "postgraphile/graphql";
 import preset from "./graphile.config.js";
 
 const pgl = postgraphile(preset);
 
-export async function executeDocument<TData = any, TVariables = any>(
+export async function executeDocument<TDoc extends TypedDocumentNode<TData, TVariables>, TData = any, TVariables = any>(
   requestContext: Partial<Grafast.RequestContext>,
-  document: DocumentNode | TypedDocumentNode<TData, TVariables>,
-  variableValues?: Record<string, unknown> | null,
+  document: TDoc,
+  variableValues?: VariablesOf<typeof document>,
   operationName?: string,
-): Promise<ExecutionResult<TData, TVariables>> {
+): Promise<ExecutionResult<ResultOf<typeof document>, TVariables>> {
```

Comment:
I don't think we need generics for TData/TVariables, maybe something like this would work:

```ts
export async function executeDocument<TDoc extends TypedDocumentNode | DocumentNode, TExtensions = ObjMap<unknown>>(
  requestContext: Partial<Grafast.RequestContext>,
  document: TDoc,
  variableValues?: VariablesOf<TDoc>,
  operationName?: string,
): Promise<ExecutionResult<ResultOf<TDoc>, TExtensions>> {
```

(Also the second parameter to ExecutionResult is TExtensions not TVariables - I recognize this was already incorrect.)